0.data_analysis/plate_eda_analyses/nbconverted/single_cell_fs_plate_cell_count_eda.py

```diff
@@ -0,0 +1,394 @@
+    _edge_well_df: Pandas Dataframe
+        Contains the cell count per edge well category and genotype
+
+    _plate_name: String
+        Name of the plate for labeling the data
+    """
+    f, ax = plt.subplots(figsize=(20, 10))
+
+    sns.barplot(data=_edge_well_df, x="Metadata_genotype", y="Metadata_number_of_singlecells", hue="Metadata_Edge_Well")
+
+    plt.ylabel('Well Cell Count')
+    plt.xlabel('Genotype')
+    plt.title('Number of cells per Genotype and Edge Well')
+    plt.savefig(plate_name_to_path_map[_plate_name] / f"{_plate_name}_genotype_cell_counts_per_well_type.png")
+    plt.close()
+
+
+# In[13]:
+
+
+def visualize_cell_count_per_site(_cell_countdf, _plate_name):
+    """
+    Visualize the cell count in a violin plot
+
+    Parameters
+    ----------
+    _cell_countdf: Pandas Dataframe
+        Conatins the cell count data
+
+    _plate_name: String
+        Name of the plate for labeling the data
+    """
+
+    f, ax = plt.subplots(figsize=(20, 10))
+
+    sns.violinplot(x=cell_count_col, y="Metadata_genotype", data=_cell_countdf)
+    plt.ylabel("Genotype")
+    plt.xlabel("Site Cell Count")
+    plt.title("Site Cell Count for each Genotype")
+    plt.savefig(plate_name_to_path_map[_plate_name] / f"{_plate_name}_violin_site_cell_counts.png")
+
+
+# In[14]:
+
+
+def visualize_cell_count_per_edge_well_genotype(_cell_countdf, _plate_name):
+    """
+    Visualize the cell count in a violin plot
+
+    Parameters
+    ----------
+    _cell_countdf: Pandas Dataframe
+        Conatins the cell count data
+
+    _plate_name: String
+        Name of the plate for labeling the data
+    """
+
+    f, ax = plt.subplots(figsize=(20, 10))
+
+    sns.violinplot(x=cell_count_col, y="Metadata_genotype", data=_cell_countdf)
+    plt.ylabel("Genotype")
+    plt.xlabel("Site Cell Count")
+    plt.title("Site Cell Count for each Genotype")
+    plt.savefig(plate_name_to_path_map[_plate_name] / f"{_plate_name}_violin_site_cell_counts.png")
+
+
+# ## Visualize Number of Missing Columns
+
+# In[15]:
+
+
+visualize_missing_columns_to_cells(plate3df, "plate_3")
+visualize_missing_columns_to_cells(plate3df, "plate_3p")
+visualize_missing_columns_to_cells(plate3df, "plate_5")
+
+
+# ### Specify Edge Wells
+
+# In[16]:
+
+
+start_letter = 'B'
+end_letter = 'G'
+
+start_num = 1
+end_num = 12
+
+edge_wells = [
+    f"{chr(wletter)}{wnum}" for wletter in range(ord(start_letter), ord(end_letter) + 1)
+    for wnum in [start_num, end_num]
+]
+
+plate3df["Metadata_Edge_Well"] = plate3df["Metadata_Well"].apply(set_edge_well)
+plate3pdf["Metadata_Edge_Well"] = plate3pdf["Metadata_Well"].apply(set_edge_well)
+plate5df["Metadata_Edge_Well"] = plate5df["Metadata_Well"].apply(set_edge_well)
+
+
+# ## Aggregate and Visualize Cell Counts
+
+# ### Cell Counts per Site
+
+# In[17]:
+
+
+site_cell_count3df = agg_cell_count_to_site(plate3df)
+site_cell_count3pdf = agg_cell_count_to_site(plate3pdf)
+site_cell_count5df = agg_cell_count_to_site(plate5df)
```

Comment:
Is there a reason why you chose to look at the site count instead of wells for these violin plots? Just curious, since I see you look at counts at the well level for edge versus inner wells! I can see the benefit of looking at both of these distributions.

Reply:
Site counts allow for higher resolution when plotting cell count distributions. For example, if there are 4 sites in a well, 3 of the sites could have 30 cells, and one site could have one cell, so aggregating to the well level would make this fact less apparent